File: crates/hyphae-server/tests/fixtures/sdk_v2_origin_free.py
from __future__ import annotations

import os

from hyphae_sdk.v2 import HyphaeClient, RequestOptions


encoded = open(os.environ["HYPHAE_ARTIFACT"], "rb").read()
offset = 0
values = []
for _ in range(3):
    length = int.from_bytes(encoded[offset:offset + 8], "little")
    offset += 8
    values.append(encoded[offset:offset + length])
    offset += length
assert offset == len(encoded)

client = HyphaeClient.http(
    os.environ["HYPHAE_ORIGIN"], bearer_token=os.environ["HYPHAE_TOKEN"]
)
verified = client.verify_proof(*values, options=RequestOptions(request_id=20_124))
assert verified.kind == "proof_verification"
assert verified.value["semantic_reexecution_performed"]

File: sdks/typescript/test/sdk-v2-origin-free.mjs
// SPDX-License-Identifier: GPL-3.0-only
import assert from "node:assert/strict";
import { readFile } from "node:fs/promises";

import { HyphaeClient } from "../dist/v2/index.js";

const encoded = new Uint8Array(await readFile(process.env.HYPHAE_ARTIFACT));
const view = new DataView(encoded.buffer, encoded.byteOffset, encoded.byteLength);
const values = [];
let offset = 0;
for (let index = 0; index < 3; index += 1) {
  const length = Number(view.getBigUint64(offset, true));
  offset += 8;
  values.push(encoded.slice(offset, offset + length));
  offset += length;
}
assert.equal(offset, encoded.byteLength);

const client = HyphaeClient.http(process.env.HYPHAE_ORIGIN, {
  bearerToken: process.env.HYPHAE_TOKEN,
});
const [proof, witness, trustedAnchor] = values;
assert.notEqual(proof, undefined);
assert.notEqual(witness, undefined);
assert.notEqual(trustedAnchor, undefined);
const verified = await client.verifyProof(proof, witness, trustedAnchor, { requestId: 30_124n });
assert.equal(verified.kind, "proof_verification");
assert.equal(verified.value.semanticReexecutionPerformed, true);
